### PR TITLE
fetch detail stats for timeRange

### PR DESCRIPTION
**Pull request by farhan-sauce**
DetailStats data was missing, which can be useful to draw following chart.

Screen Shot 2019-09-17 at 7 31 02 PM

### CODE REVIEW

File: tests/index.test.ts [hunk 1]
```diff
@@ -62,3 +62,20 @@ describe('mainTrackEvents', () => {
         expect(result.length).toEqual(56244)
     })
 })
+
+describe('getDetailStats', () => {
+    it('should get detail stats', () => {
+        const result = trace.getDetailStats()
+        expect(Object.keys(result)).toMatchSnapshot()
```
Comment (christian-bromann) [inline tests/index.test.ts:69]: Can we also test the values instead just keys?

File: tests/index.test.ts [hunk 2]
```diff
@@ -62,3 +62,20 @@ describe('mainTrackEvents', () => {
...
+    })
+
+    it('should not throw error on second call of getDetailStats', () => {
+        const result = trace.getDetailStats()
+        expect(Object.keys(result)).toMatchSnapshot()
```
Comment (christian-bromann) [inline tests/index.test.ts:74]: same

File: tests/index.test.ts [hunk 3]
```diff
@@ -62,3 +62,20 @@ describe('mainTrackEvents', () => {
...
+    })
+
+    it('should get summary data between passed range', () => {
+        const result = trace.getDetailStats(289960055.634, 289960729.717)
+        expect(Object.keys(result)).toMatchSnapshot()
```
Comment (christian-bromann) [inline tests/index.test.ts:79]: same

**Review by christian-bromann — CHANGES_REQUESTED** [review]

File: src/utils.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
+import TimelineUIUtils from './../devtools/timelineModel/timelineUIUtils'
+import { StatsArray } from './../devtools/types'
+import TimelineModel from './../devtools/timelineModel'
+import Event from './../devtools/tracingModel/event'
+import TracingModel from './../devtools/tracingModel'
+
+export default class CustomUtils extends TimelineUIUtils {
+    public constructor() {
+        super()
```
Comment (christian-bromann) [inline src/utils.ts]: the constructor function is not needed if you just call `super()` in it.

**Review by christian-bromann — CHANGES_REQUESTED** [review]

File: tests/index.test.ts (hunk 3 above)
Comment (farhan-sauce) [reply tests/index.test.ts:79]: `getDetailStats` returns big object ,it takes 234k lines in snapshot file, is it okay?

Comment (christian-bromann) [reply tests/index.test.ts:79]: no, let's keep it this way then

**Review by christian-bromann — APPROVED** [review]

**Pull request merged**